src/module.c

```diff
@@ -10987,10 +11123,429 @@ sds genModulesInfoString(sds info) {
+}
+
+unsigned int maskModuleNumericConfigFlags(unsigned int flags) {
+    unsigned int new_flags = 0;
+    if (flags & REDISMODULE_CONFIG_MEMORY) new_flags |= MEMORY_CONFIG;
+    return new_flags;
+}
+
+int moduleVerifyConfigName(sds name) {
+    if (sdslen(name) == 0) {
+        return 1;
+    }
+    for (size_t i = 0 ; i < sdslen(name) ; ++i) {
+        char curr_char = name[i];
+        if ((curr_char >= 'a' && curr_char <= 'z') ||
+            (curr_char >= 'A' && curr_char <= 'Z') ||
+            (curr_char >= '0' && curr_char <= '9') ||
+            (curr_char == '_') || (curr_char == '-'))
+        {
+            continue;
+        }
+        return 1;
+    }
+    return 0;
+}
+
+/* Create a bool config that server clients can interact with via the 
+ * `CONFIG SET`, `CONFIG GET`, and `CONFIG REWRITE` commands. See 
+ * RedisModule_RegisterStringConfig for detailed information about configs. */
+int RM_RegisterBoolConfig(RedisModuleCtx *ctx, const char *name, int default_val, unsigned int flags, RedisModuleConfigGetBoolFunc getfn, RedisModuleConfigSetBoolFunc setfn, RedisModuleConfigApplyFunc applyfn, void *privdata) {
+    RedisModule *module = ctx->module;
+    if (moduleConfigValidityCheck(module, name, flags, BOOL_CONFIG)) {
+        return REDISMODULE_ERR;
+    }
+    sds config_name = sdsnew(name);
+    if (moduleVerifyConfigName(config_name)) {
+        return REDISMODULE_ERR;
+    }
+    ModuleConfig *new_config = createModuleConfig(config_name, applyfn, privdata, module);
+    new_config->get_fn.get_bool = getfn;
+    new_config->set_fn.set_bool = setfn;
+    listAddNodeTail(module->module_configs, new_config);
+    flags = maskModuleConfigFlags(flags);
+    addModuleBoolConfig(module->name, name, flags, new_config, default_val);
+    return REDISMODULE_OK;
+}
+
+/* Create a string config that server clients can interact with via the 
+ * `CONFIG SET`, `CONFIG GET`, and `CONFIG REWRITE` commands.
+ * 
+ * The actual config value is owned by the module, with GET, SET and optional APPLY callbacks
+ * that are provided to Redis in order to access or manipulate the value. The 
+ * GET callback retrieves the value from the module, while the SET
+ * callback provides a value to be stored into the config. The optional APPLY callback
+ * is called after a CONFIG SET command and can be used to apply a config after it and potentially other
+ * configs were changed atomically. If there are multiple apply callbacks specified in
+ * a CONFIG SET command, they will be deduplicated if the function and privdata pointers are the same,
+ * and it will only be run once. The config also declares a type for the value that is validated by Redis
+ * and provided to the module. The config system provides the following types:
+ * 
+ * * Redis String: Binary safe string data.
+ * * Enum: One of a finite number of string tokens, provided during registration.
+ * * Integer: 64 bit signed integer, which also supports min and max values.
```

Comment:
We call this an integer, but everything else is called numeric. Let's make it consistent.